vignettes/ipums-api.Rmd

```diff
@@ -0,0 +1,325 @@
+)
+```
+
+The `data_format`, `data_structure`, and `rectangular_on` arguments have default 
+values which match the values specified in the example above, and you can omit 
+the argument names as long as you maintain the proper order, so you could define 
+the same extract with:
+
+```{r}
+extract_definition <- define_extract(
+  "usa",
+  "Extract for API vignette",
+  c("us2018a","us2019a"),
+  c("AGE","SEX","RACE","STATEFIP")
+)
+```
+
+
+# Submitting your extract
+
+Submitting your extract is as easy as:
+
+```{r eval=FALSE}
+submit_extract(extract_definition)
+```
+
+
+However, like the `define_extract()` function, the `submit_extract()` function
+returns an "ipums_extract" object, and the returned object has been updated to
+include the extract number, so it can be convenient to save that return object
+by assigning a name to it, like this:
+
+```{r}
+submitted_extract <- submit_extract(extract_definition)
+```
+
+That way, you can use the `submitted_extract` object as input to check the
+extract's status, as shown in the next section.
+
+
+# Checking the status of your extract
+
+To retrieve the latest status of an extract, you can use the
+`get_extract_info()` function. `get_extract_info()` returns an "ipums_extract" 
+object with the "status" element updated to reflect the latest 
+status of the extract, and the "download_links" element updated to include links 
+to any extract files that are available for download. The "status" of an extract 
+is one of "queued", "started", "produced", "canceled", "failed", or "completed". 
+
+If you assigned a name to the return value of `submit_extract()`, as shown
+above, you could get update the "ipums_extract" object with:
+
+```{r}
+submitted_extract <- get_extract_info(submitted_extract)
+```
+
+To print the latest status, you can use:
+
+```{r}
+submitted_extract$status
+```
+
+If you don't have an "ipums_extract" object in your environment that describes 
+the extract you're interested in, you can also query the latest status of an 
+extract by supplying the name of the IPUMS data collection and extract number of 
+the extract, in one of two formats:
+
+```{r eval=FALSE}
+submitted_extract <- get_extract_info("usa:33")
+```
+
+Or:
+
+```{r eval=FALSE}
+submitted_extract <- get_extract_info(c("usa", "33"))
+```
+
+If you want R to periodically check the status of your extract, and only return 
+an updated "ipums_extract" object once the extract is ready to download, you 
+can use `wait_for_extract()`, as shown below:
+
+```{r eval=FALSE}
+downloadable_extract <- wait_for_extract(submitted_extract)
+```
+
+`wait_for_extract()` also accepts the same `"collection:number"` and 
+`c("collection", "number")` specifications shown above:
+
+```{r eval=FALSE}
+downloadable_extract <- wait_for_extract("usa:33")
+```
+
+Or:
+
+```{r eval=FALSE}
+downloadable_extract <- wait_for_extract(c("usa", "33"))
+```
+
+For large extracts that take a long time to produce, you may not want to use 
```

Comment:
add "or when the IPUMS servers are busy"?

Reply:
Added this, thanks.